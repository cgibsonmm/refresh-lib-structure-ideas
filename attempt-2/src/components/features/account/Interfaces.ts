import { Nullable } from '@/utils/helperTypes';
export interface AccountResponse {
  account: Account;
  meta: Meta;
}

interface Meta {
  status: number;
  free_user: boolean;
  mobile_free: boolean;
  subscription_state: string;
  notification?: SiteNotification | null;
}
interface SiteNotification {
  message: string;
  title: string;
}

export interface Account {
  active_ab_tests: ActiveAbTests;
  alert_me: AlertMe;
  billing_info: BillingInfo;
  legal_doc_info: LegalDocInfo;
  notification?: AccountNotification | null;
  report_tags: ReportTags;
  retention_info: RetentionInfo;
  sandboxed: boolean;
  search_counts: SearchCounts;
  staff_info: StaffInfo;
  subscription_info: SubscriptionInfo;
  upgrade_info: UpgradeInfo;
  user_info: UserInfo;
  user_settings: UserSettings;
}

export interface UserInfo {
  first_name: string | null;
  last_name: string | null;
  email: string;
  phone_number: string;
  optout_options: boolean;
  user_code: string;
  optout_deleted: boolean;
  join_date: string;
  signup_type: string;
  design_name: string;
  design_version: string;
  tos_version_bad: boolean;
  pp_version_bad: boolean;
  noncompliant: boolean;
  tos_pending: boolean;
  'shut_down?': boolean;
  hide_person_video: boolean;
  hide_property_video: boolean;
  friends_and_family: boolean;
  user_in_recycling: boolean;
  recycling_date: string | null;
  is_self_searcher: boolean;
  seats: Seat[];
  seat_status: string | null;
  invite_count: number;
  has_connected_vehicles: boolean;
}

export interface SubscriptionInfo {
  free_user: boolean;
  'mobile_free?': boolean;
  public_plan_type: string;
  subscription_plan_unique_name: string;
  subscription_plan_internal_name: string;
  subscription_state: string;
  in_billable_state: boolean;
  recurring: boolean;
  public_price_description: string;
  normalize_date: string;
  monitor_limit: number;
  remaining_monitors: number;
  credits_remaining: number;
  monthly_reports_remaining: number;
  monthly_report_limit: number;
  recycling_reports_remaining: number;
  subscription_features: SubscriptionFeatures;
  money_amount: string;
  'show_unlimited_reports?': boolean;
  renewal_period: number;
  renewal_period_type: string;
}

export type SubscriptionFeatures = Nullable<{
  monthly_report_limit: number;
  consumer_report_block_plan_name: string;
  unlimited_reports: boolean;
  unlimited_monitors: boolean;
  report_tier: string;
  monitor_limit: number;
  daily_report_soft_limit: number;
  monthly_bjl_limit: number;
  comp_reports_enabled: boolean;
  alertme_enabled: boolean;
  unlimited_bankruptcies: boolean;
  unlimited_liens_and_judgments: boolean;
  lien_and_judgment_limit: number;
  six_month_transaction_multiplier: number;
  default_design: string;
  default_search_bar: string;
  modal_family: string;
  bankruptcy_limit: number;
  session_lock_enabled: boolean;
  pdf_access: boolean;
  pro_account: boolean;
  hourly_report_soft_limit: number;
  active_reports_limit: number;
  api_access: boolean;
}>;

export interface SubscriptionPlan {
  amazon_id: string | null;
  amount: number;
  app_only: string;
  apple_id: string | null;
  google_id: string | null;
  name: string;
  title: string;
  subtitle: string;
  recurring: boolean;
  renewal_period: number;
  renewal_period_type: string;
  report_tier: string;
  unique_key: string;
  internal_name: string;
  discount: string | null;
  public_price_description: string;
  default: boolean;
  monthly_report_limit: number;
}

export interface UpgradeInfo {
  upgrade_credits: number;
}

export interface SearchCounts {
  replenishment_date: string;
  allocated_bankruptcies: number;
  allocated_liens_and_judgments: number;
  remaining_bankruptcies: number;
  remaining_liens_and_judgments: number;
}

export interface LegalDocInfo {
  active_tos_version: number;
  active_privacy_policy_version: number;
}

export interface BillingInfo {
  card_type?: string;
  name?: string;
  card_number?: string;
  street?: string;
  city?: string;
  state?: string;
  postal_code?: string;
  country?: string;
}

export interface AlertMe {
  has_access: boolean;
  settings: AlertMeSettings;
}

export interface AlertMeSettings {
  id: number;
  digest_enabled: boolean;
  push_digest_enabled: boolean;
  instant_enabled: boolean;
  push_instant_enabled: boolean;
  next_digest_mailing_at: string;
  created_at: string;
  updated_at: string;
}

export type RetentionInfo = Nullable<{
  ran_report_on_self: boolean;
  ran_report_on_multiple_days: boolean;
  ran_report_on_family_member: boolean;
  ran_reverse_phone_search: boolean;
  ran_email_search: boolean;
  ran_professional_contact_search: boolean;
  ran_people_search: boolean;
  ran_property_search: boolean;
  ran_estate_value_search: boolean;
  created_monitor: boolean;
  created_monitor_on_self: boolean;
  created_monitor_on_family_member: boolean;
  created_monitor_on_property: boolean;
  created_monitor_on_email: boolean;
  created_monitor_on_their_email: boolean;
  has_used_mobile_app: boolean;
  has_used_website: boolean;
  downsell_plan_offered: boolean;
  downsell_plan_accepted: boolean;
  claimed_person_report: boolean;
  ran_username_search: boolean;
  ran_neighborhood_watch_search: boolean;
  ran_vehicle_search: boolean;
  claimed_report: boolean;
}>;

export type ReportTags = Nullable<{
  self_person_report_id: string;
}>;

export interface ActiveAbTests {
  tests: ActiveAbTest[];
}

export interface ActiveAbTest {
  id: number;
  user_id: number;
  source: string;
  experiment_key: string;
  variation_key: string;
  created_at: string;
  updated_at: string;
  whitelist: string | null;
}
export type UserSettings = Nullable<{
  hide_reactivation_banner: boolean;
  hide_recurring_cancelled_banner: boolean;
  hide_bad_billing_banner: boolean;
  hide_credit_card_banner: boolean;
  hide_trial_welcome_banner: boolean;
  skip_creation_modal: boolean;
  hide_person_video: boolean;
  hide_property_video: boolean;
  hide_dark_web_banner: boolean;
  hide_dark_web_notifications: boolean;
  hide_unclaimed_money_banner: boolean;
  hide_unclaimed_money_ctas: boolean;
  hide_download_app_cta: boolean;
  hide_unclaimed_money_welcome_modal: boolean;
  hide_social_cleanup_cta: boolean;
  show_info_collection_screen: boolean;
  hide_how_bv_works_modal: boolean;
  hide_payment_update_required_modal: boolean;
  hide_onboarding_checklist_modal: boolean;
  show_higher_confidence_data_only: boolean;
  show_sales_tax_notice_modal: boolean;
  hide_rate_app_banner: boolean;
  hide_multi_seats_invite_modal: boolean;
  hide_criminal_records: boolean;
  hide_privacy_cta: boolean;
  hide_ps_vs_biz_modal: boolean;
  hide_own_fcra: boolean;
  hide_own_tos: boolean;
  show_section_learn_more: boolean;
  hide_cancel_downsell_modal: boolean;
  check_marketplace_step: boolean;
  dqs_completed: boolean;
  hide_passes_invite_modal: boolean;
  hide_sitejabber_widget: boolean;
  check_invites_step: boolean;
  check_extension_step: boolean;
  check_reward_1_step: boolean;
  check_reward_2_step: boolean;
  check_reward_3_step: boolean;
  hide_getting_started: boolean;
  high_confidence_toggle: boolean;
  check_list_completed: boolean;
  check_list_skipped: boolean;
  identity_hub_completed: boolean;
  tips_completed: boolean;
  first_report_modal_seen: boolean;
  openbay_callout_modal_seen: boolean;
  credit_hub_enrollment_completed: boolean;
}>;

export interface StaffInfo {
  admin: boolean;
  blacklister: boolean;
  staff: boolean;
  beta_tester: boolean;
}

interface Seat {
  id: string;
  email: string;
  first_name: string;
  last_name: string;
  status: string;
  created_at: string;
}

export interface AccountNotification {
  type: string;
  title: string;
  message: string;
  call_to_action: string;
}

//Payments
interface ReceiptItem {
  amount: number | string;
  description: string;
  id: number;
  purchaseable_type: string;
  receipt_id: number;
}

export interface Receipt {
  amount: number | string;
  created_at: string;
  id: number;
  method: string;
  items: ReceiptItem[];
}

export interface PaymentMethods {
  active: boolean;
  card_type: string;
  city: string;
  country: string;
  expiration_month: number;
  expiration_year: number;
  email: string;
  first_name: string;
  id: number;
  last_four: string;
  last_name: string;
  masked_card_number: string;
  payer_id: string;
  state: string;
  street1: string;
  payer_status: string;
  postal_code: string;
  description: string;
  braintree_type: string;
  token: string;
  payment_type: string;
}
export interface PaymentImages {
  master: string;
  visa: string;
  american_express: string;
  discover: string;
  genericCreditCard: string;
  paypal: string;
  applePay: string;
  googlePay: string;
  venmo: string;
}
