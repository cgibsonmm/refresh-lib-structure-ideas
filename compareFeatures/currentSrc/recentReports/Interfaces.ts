import { Nullable } from '@/utils/helperTypes';

export type RawReportType =
  | 'social_network_report'
  | 'detailed_person_report'
  | 'username_report'
  | 'reverse_phone_report'
  | 'sex_offender_report'
  | 'property_report'
  | 'detailed_business_report';

export interface RecentReportRawData {
  age?: number;
  address?: string;
  alert_me?: boolean | null;
  city?: string;
  created_at?: string;
  email?: string;
  first_name?: string;
  full_size_image?: string | null;
  has_new_info?: boolean;
  interaction?: Record<string, unknown>;
  license_plate?: string;
  last_name?: string;
  make?: string;
  middle_name?: string | null;
  mileage?: number;
  model?: string;
  permalink: string;
  phone_number?: string;
  rating?: unknown;
  report_tier?: string;
  report_type: RawReportType;
  report_upgraded?: boolean;
  state?: string;
  search_by?: string;
  service_data?: unknown;
  tag?: string | null;
  thumb?: unknown;
  updated_at?: string;
  upgrade_options?: unknown;
  vin?: string;
  year?: number;
  username?: string;
  network?: string;
  zip_code?: string;
  domain?: string;
  no_entities?: unknown;
  carrier?: string;
  bvids?: string[];
}
export interface RecentReportsRawData {
  meta: ReportsMeta;
  polling: Record<string, unknown>;
  reports: RecentReportRawData[];
}

export interface RecentReportsDataSerialized {
  reports: RecentReportSerialized[];
  meta: ReportsMeta;
  polling: Record<string, unknown>;
}

export interface RecentReportSerialized {
  id: string;
  reportUpgraded?: boolean;
  createdAt: string;
  createdDate: string;
  updatedAt: string;
  updatedDate: string;
  alertMe?: boolean | null;
  hasNewInfo: boolean | null;
  reportType: string;
  reportTypeFormatted: string;
  detailsCityState?: string;
  reportData: string;
  details: RecentReportRawData;
}

interface ReportsMeta {
  status: number;
  free_user: boolean;
  mobile_free: boolean;
  subscription_state: string;
  report_quantities: Nullable<ReportQuantities>;
}

interface ReportQuantities {
  contact_reports_count: number;
  detailed_business_reports_count: number;
  email_reports_count: number;
  estate_value_reports_count: number;
  filtered_reports_count: number;
  fraud_reports_count: number;
  link_to_next_page: number;
  link_to_prev_page: number;
  person_reports_count: number;
  phone_reports_count: number;
  property_reports_count: number;
  report_limit: number;
  sex_offender_reports_count: number;
  total_reports_count: number;
  username_reports_count: number;
  vehicle_reports_count: number;
}
