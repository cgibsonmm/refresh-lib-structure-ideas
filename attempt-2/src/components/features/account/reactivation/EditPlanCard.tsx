import { Text } from '@/theme/index';
import { dollarize } from '@/utils/dollarize';
import { Left, PlanCardContainer, PlanCheckbox, Right } from './components';
import { SubscriptionPlan } from '../Interfaces';

export function EditPlanCard({
  plan,
  selectedPlan,
  handleSelectedPlan,
}: {
  /** Current user's plan. */
  plan: SubscriptionPlan;
  /** The plan selected by the user to reactivate. */
  selectedPlan: SubscriptionPlan;
  /** Callback fired when the user selects a plan. */
  handleSelectedPlan: (
    plan_internal_name: string,
    selectedPLanMonthlyReportLimit: number
  ) => void;
}) {
  const { title, subtitle, amount, internal_name, monthly_report_limit } = plan;

  const handleClick = () => {
    handleSelectedPlan(internal_name, monthly_report_limit);
  };

  return (
    <PlanCardContainer
      onClick={handleClick}
      role="button"
      key={plan.internal_name}
    >
      <PlanCheckbox
        checked={selectedPlan.internal_name === plan.internal_name}
      />
      <Left>
        <Text variant="h3">{title}</Text>
        <Text
          variant="body1"
          color="text.secondary"
          dangerouslySetInnerHTML={{ __html: subtitle }}
        />
      </Left>
      <Right>
        <Text variant="h2">{dollarize(amount)}</Text>
        <Text color="text.secondary" sx={{ textAlign: 'right' }}>
          Per month plus
        </Text>
        <Text
          variant="body1"
          color="text.secondary"
          sx={{ textAlign: 'right' }}
        >
          applicable sales tax
        </Text>
      </Right>
    </PlanCardContainer>
  );
}
