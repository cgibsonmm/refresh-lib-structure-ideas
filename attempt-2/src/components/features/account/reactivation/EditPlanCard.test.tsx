import { render, fireEvent } from '@testing-library/react';
import { EditPlanCard } from './EditPlanCard';
import { SubscriptionPlan } from '../Interfaces';

const mockPlan = {
  title: 'Basic Plan',
  subtitle: 'Includes basic features',
  amount: 10000,
  internal_name: 'basic_plan',
};

const mockSelectedPlan = {
  title: 'Premium Plan',
  subtitle: 'Includes premium features',
  amount: 20000,
  internal_name: 'premium_plan',
};

const emptyFn = jest.fn();

describe('EditPlanCard', () => {
  it('renders', () => {
    const { asFragment } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockSelectedPlan as SubscriptionPlan}
        handleSelectedPlan={emptyFn}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the title and subtitle correctly', () => {
    const { getByText } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockSelectedPlan as SubscriptionPlan}
        handleSelectedPlan={emptyFn}
      />
    );
    expect(getByText('Basic Plan')).toBeInTheDocument();
    expect(getByText('Includes basic features')).toBeInTheDocument();
  });

  it('renders the amount correctly', () => {
    const { getByText } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockSelectedPlan as SubscriptionPlan}
        handleSelectedPlan={emptyFn}
      />
    );
    expect(getByText('$100')).toBeInTheDocument();
  });

  it('renders the "Per month" text', () => {
    const { getByText } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockSelectedPlan as SubscriptionPlan}
        handleSelectedPlan={emptyFn}
      />
    );
    expect(getByText('Per month plus')).toBeInTheDocument();
  });

  it('renders the sales tax text correctly', () => {
    const { getByText } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockSelectedPlan as SubscriptionPlan}
        handleSelectedPlan={emptyFn}
      />
    );
    expect(getByText('applicable sales tax')).toBeInTheDocument();
  });

  it('calls the handleSelectedPlan function when clicked', () => {
    const mockHandleSelectedPlan = jest.fn();
    const { getByRole } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockSelectedPlan as SubscriptionPlan}
        handleSelectedPlan={mockHandleSelectedPlan}
      />
    );
    fireEvent.click(getByRole('button'));
    expect(mockHandleSelectedPlan).toHaveBeenCalledWith(
      'basic_plan',
      undefined
    );
  });

  it('renders the checkbox as checked if the plan is selected', () => {
    const { getByRole } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockPlan as SubscriptionPlan}
        handleSelectedPlan={emptyFn}
      />
    );
    const checkbox = getByRole('checkbox');
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).toBeChecked();
  });

  it('renders the checkbox as unchecked if the plan is not selected', () => {
    const { getByRole } = render(
      <EditPlanCard
        plan={mockPlan as SubscriptionPlan}
        selectedPlan={mockSelectedPlan as SubscriptionPlan}
        handleSelectedPlan={emptyFn}
      />
    );
    const checkbox = getByRole('checkbox');
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
  });
});
