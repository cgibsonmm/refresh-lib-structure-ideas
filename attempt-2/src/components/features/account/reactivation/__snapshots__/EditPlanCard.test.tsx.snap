// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPlanCard renders 1`] = `
<DocumentFragment>
  <div
    class="css-kksclf"
    role="button"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-tque03-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <div
      class="css-1lxwves"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      >
        Basic Plan
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
      >
        Includes basic features
      </p>
    </div>
    <div
      class="css-1ys4hkf"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        $100
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 css-17hfm4p-MuiTypography-root"
      >
        Per month plus
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-17hfm4p-MuiTypography-root"
      >
        applicable sales tax
      </p>
    </div>
  </div>
</DocumentFragment>
`;
